src/utils/stringifyVariables.ts

```diff
@@ -1,6 +1,7 @@
 const seen = new Set();
 
 const stringify = (x: any): string => {
+  if (x && x.toJson) x = x.toJSON();
```

Comment:
Seems like this isn’t quite the right check. We could also move it down to avoid the double null check and do a proper `typeof` being “function” check?

I’m also wondering, what are we really accomplishing for moment here? Don’t we want to convert it and dates to a UTC string?

Reply:
If I recall correctly the normal way for serializing a complex object is through the `.toJSON` function which is present on most libraries like `moment`, `day-js`,...

Reply:
When writing the test I found out that it doesn't work this way, because for some reason `x.toJson` is undefined. At least for javascript dates `JSON.stringify(x)` works though